TypeScriptDefinitions/async.d.ts

```diff
@@ -0,0 +1,165 @@
+// Type definitions for Async 1.4.2
```

Comment:
I saw that it was included in some of our other repos, @mramato should it be included?